### ISSUE TITLE
Validation for node names

### ISSUE THREAD

**Issue by VermaSh** [AdoptOpenJDK/openjdk-jenkins-helper#3]
Once we have figured out a naming convention for nodes we should implement a validation function in the API. As of right now it only checks for length greater than 2, it didn't make sense to check for just 0 or 1.

**Comment by gdams** [comment]
So the naming convention has been agreed: `build-cloudcone-ubuntu1604-x64-1`

**Comment by jdekonin** [comment]
Agreed by who?  Is there a link to a previous discussion?  Does this mean any machine that name build-* cannot run testing of any kind?

**Comment by gdams** [comment]
https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/162. The name is generated from our inventory.yml.

**Comment by jdekonin** [comment]
As long as the tooling is written in a manner that the naming isn't enforced for all environments.  For labs that might only have a small number of machines having a machine exclusively build or test isn't practical.  Example Eclipse/OMR or Eclipse/OpenJ9 cis for which this plugin could also be used.  Is there a category or keyword for both test and build?
